v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/RangeCountDoFn.java

```diff
@@ -97,43 +103,49 @@ public void processElement(@Element Range input, OutputReceiver<Range> out, Proc
               this.countQuery, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
       stmt.setQueryTimeout((int) ((this.timeoutMillis + TIMEOUT_GRACE_MILLIS) / 1000));
       new RangePreparedStatementSetter(numColumns).setParameters(input, stmt);
-      try {
-        ResultSet rs = stmt.executeQuery();
-        if (rs.next()) {
-          count = rs.getLong(1);
-          if (rs.wasNull()) {
-            logger.error(
-                "Got null for count resultSet. Range = {}, Query = {}, DataSource = {}",
-                input,
-                countQuery,
-                dataSource);
-            count = Range.INDETERMINATE_COUNT;
-          }
-        } else {
+      ResultSet rs = stmt.executeQuery();
+      if (rs.next()) {
+        count = rs.getLong(1);
+        if (rs.wasNull()) {
           logger.error(
-              "Got empty count resultSet. Range = {}, Query = {}, DataSource = {}",
+              "Got null for count resultSet. Range = {}, Query = {}, DataSource = {}",
               input,
               countQuery,
               dataSource);
           count = Range.INDETERMINATE_COUNT;
         }
-      } catch (SQLTimeoutException e) {
+      } else {
+        logger.error(
+            "Got empty count resultSet. Range = {}, Query = {}, DataSource = {}",
+            input,
+            countQuery,
+            dataSource);
+        count = Range.INDETERMINATE_COUNT;
+      }
+    } catch (SQLException e) {
+      if (checkTimeout(e)) {
         logger.warn(
-            "Handled timeout while counting Range = {}, Query = {}, DataSource = {}, timeoutMillis = {}",
+            "Handled timeout while counting Range = {}, Query = {}, DataSource = {}, timeoutMillis = {}, Exception {}, sqlState {}, errorCode {}",
             input,
             countQuery,
             dataSource,
-            timeoutMillis);
+            timeoutMillis,
+            e,
+            e.getSQLState(),
+            e.getErrorCode());
         count = Range.INDETERMINATE_COUNT;
+      } else {
+        logger.error(
+            "Non-timeout SQL Exception while counting Range = {}, Query = {}, DataSource = {}, timeoutMillis = {}, exception {}, sqlState {}, errorCode {}",
+            input,
+            countQuery,
+            dataSource,
+            timeoutMillis,
+            e,
+            e.getSQLState(),
+            e.getErrorCode());
+        throw e;
```

Comment:
could this be a temporary connection issue or something ? In which case we should not fail the splitter as this is best effort. We should fall back to undetermined count.

Reply:
we could do that, but currently all the exceptions we throw at run time are retried by beam, it's the same strategy we follow for read too.

Reply:
Add a todo to re-evaluate this with test runs.